src/Illuminate/Http/TransformableResponse.php

```diff
@@ -0,0 +1,335 @@
+                    );
+                }
+
+                $parsedRules[$rule] = $rules[$rule];
+
+                return $parsedRules;
+            }, []);
+    }
+
+    /**
+     * Sanitize rules removing those that don't appear into orginal rules.
+     *
+     * @param  string  $rule
+     * @param  array  $gatheredRules
+     * @param  bool  $valueForValidOnes
+     * @return array
+     */
+    protected function sanitizeArrayGatheredRules($rule, array $gatheredRules, $valueForValidOnes)
+    {
+        $pattern = '/'.str_replace('.*.', '\.([0-9])+\.', $rule).'/';
+
+        return array_reduce($gatheredRules,
+            function ($validRules, $rule) use ($pattern, $valueForValidOnes) {
+                preg_match($pattern, $rule, $matches);
+
+                if ($matches) {
+                    $validRules[$matches[0]] = $valueForValidOnes;
+                }
+
+                return $validRules;
+            },
+        []);
+    }
+
+    /**
+     * Apply visbility rules to given data.
+     *
+     * @param  array  $rules
+     * @return array
+     */
+    protected function applyVisibilityRules(array $data, array $rules)
+    {
+        if (empty($rules)) {
+            return $data;
+        }
+
+        $data = $this->showFields($data, $rules);
+        $data = $this->hideFields($data, $rules);
+
+        return $data;
+    }
+
+    /**
+     * Apply rules over the fields that must be displayed.
+     *
+     * @param  array  $data
+     * @param  array  $rules
+     * @return array
+     */
+    protected function showFields(array $data, array $rules)
+    {
+        $applicableRules = array_filter($rules);
+
+        if (empty($applicableRules)) {
+            return $data;
+        }
+
+        return array_reduce(array_keys($applicableRules),
+            function ($transformedData, $attribute) use ($data) {
+                if ($value = Arr::get($data, $attribute)) {
+                    Arr::set(
+                        $transformedData,
+                        $this->applyRenamingRules($attribute),
+                        $this->applyMutationRules(
+                            $attribute,
+                            $this->applyCastingRules($attribute, $value)
+                        )
+                    );
+                }
+
+                return $transformedData;
+            },
+        []);
+    }
+
+    /**
+     * Apply rules over the fields that must be hidden.
+     *
+     * @param  array  $data
+     * @param  array  $rules
+     * @return array
+     */
+    protected function hideFields(array $data, array $rules)
+    {
+        $applicableRules = array_filter($rules, function ($rule) {
+            return ! $rule;
+        });
+
+        if (empty($applicableRules)) {
+            return $data;
+        }
+
+        return array_reduce(array_keys($applicableRules),
+            function ($transformedData, $rule) {
+                if (Arr::has($transformedData, $rule)) {
```

Comment:
`\Illuminate\Support\Arr::forget` always make sure that the array key exist, so using `\Illuminate\Support\Arr::has` is useless.